src/serac/physics/utilities/finite_element_state.hpp

```diff
@@ -163,17 +173,92 @@ class FiniteElementState {
   {
     static_assert(std::is_constructible_v<Tensor, mfem::ParFiniteElementSpace*>,
                   "Tensor must be constructible with a ptr to ParFESpace");
-    return std::make_unique<Tensor>(&space_);
+    return std::make_unique<Tensor>(&retrieve(space_));
   }
 
 private:
+  /**
+   * @brief A helper type for uniform semantics over owning/non-owning pointers
+   */
+  template <typename T>
+  using MaybeOwner = std::variant<T*, std::unique_ptr<T>>;
+
+  /**
+   * @brief Retrieves a reference to the underlying object in a MaybeOwner
+   * @param[in] obj The object to dereference
+   */
+  template <typename T>
+  static T& retrieve(MaybeOwner<T>& obj)
+  {
+    return std::visit([](auto&& ptr) -> T& { return *ptr; }, obj);
+  }
+  /// @overload
+  template <typename T>
+  static const T& retrieve(const MaybeOwner<T>& obj)
+  {
+    return std::visit([](auto&& ptr) -> const T& { return *ptr; }, obj);
+  }
+
   // Allows for copy/move assignment
-  std::reference_wrapper<mfem::ParMesh>          mesh_;
-  std::unique_ptr<mfem::FiniteElementCollection> coll_;
-  mfem::ParFiniteElementSpace                    space_;
-  std::unique_ptr<mfem::ParGridFunction>         gf_;
-  mfem::HypreParVector                           true_vec_;
-  std::string                                    name_ = "";
+  mfem::ParMesh* mesh_;
+  // Must be const as FESpaces store a const reference to their FEColls
+  MaybeOwner<const mfem::FiniteElementCollection> coll_;
+  MaybeOwner<mfem::ParFiniteElementSpace>         space_;
+  mfem::ParGridFunction*                          gf_;
+  mfem::HypreParVector                            true_vec_;
+  std::string                                     name_ = "";
+};
+
+/**
+ * @brief Manages the lifetimes of FEState objects such that restarts are abstracted
+ * from physics modules
+ */
+class StateManager {
+public:
+  /**
+   * @brief Initializes the StateManager with a sidre DataStore (into which state will be written/read)
+   * @param[in] ds The DataStore to use
+   * @param[in] cycle_to_load The cycle to load - required for restarts
+   */
+  static void initialize(axom::sidre::DataStore& ds, const std::optional<int> cycle_to_load = {});
+
+  /**
+   * @brief Factory method for creating a new FEState object, signature is identical to FEState constructor
+   * @param[in] options Configuration options for the FEState, if a new state is created
+   * @see FiniteElementState::FiniteElementState
+   * @note If this is a restart then the options (except for the name) will be ignored
+   */
+  static FiniteElementState newState(FiniteElementState::Options&& options = {});
+
+  /**
+   * @brief Updates the Conduit Blueprint state in the datastore and saves to a file
+   * @param[in] t The current sim time
+   * @param[in] cycle The current iteration number of the simulation
+   */
+  static void step(const double t, const int cycle);
```

Comment:
It's a little confusing to me why a method named `step` actually saves to a file. Maybe `save`?